airqo-platform/AirQo-api
# [Analytics] final refactoring

#### Summary of Changes (What does this PR do?)
- add exceedance endpoint
- add averages endpoint
- cache dashboard endpoints
- remove old code

#### Is this change ready to hit production in its current state?

#### Status of maturity (all need to be checked before merging):


#### How should this be manually tested?
* Fetch and checkout to this branch locally
* Change directory to the analytics microservice
* Start redis (in another terminal)
* Create and/or start your `python` virtual env
* Install requirements `pip install -r requirements.txt
* Set the `.env` file. Sample [here](https://airqo.slack.com/archives/C0278FK2EGG/p1626124999013700)
* Start application `flask run`
* Check the swagger docs `http://localhost:5000/apidocs/`
* Test the exceedances endpoint

#### What are the relevant tickets?
- [PLAT-658](https://airqoteam.atlassian.net/browse/PLAT-658)
